services/graph/pkg/service/v0/utils.go

```diff
@@ -454,3 +458,19 @@ func roleConditionForResourceType(ri *storageprovider.ResourceInfo) (string, err
 		return "", errorcode.New(errorcode.InvalidRequest, "unsupported resource type")
 	}
 }
+
+// GetShareID returns a valid share ID from the given resource ID.
```

Comment:
Why do we know that it returns a valid share id? I most cases the `OpaqueId` of a ResourceId does not represent a valid share id.